Source/WebCore/platform/graphics/filters/Filter.cpp

```diff
@@ -102,4 +103,21 @@ RefPtr<FilterImage> Filter::apply(ImageBuffer* sourceImage, const FloatRect& sou
     return result;
 }
 
+void Filter::applyStyles(GraphicsContext& context) const
+{
+    for (auto& style : createStyles()) {
+        context.save();
```

Comment:
Man these disconnected save and restores make me nervous. I wonder if there is a way we can ensure these are always both invoked statically?